### ISSUE TITLE
socialSharing shareViaEmail returned promise doesn't trigger.

### ISSUE THREAD

**Issue by mhateo** [driftyco/ionic-native#518]
Hello,
I'm using ionic native version 1.3.17 and I'm experiencing some problems in the returned promise of the shareViaEmail method of the socialSharing plugin.

The problem is that neither then nor catch functions are getting called, so I can't know if and when the user has sent an email.

canShareViaEmail and shareWithOptions methods work as expected, but I can't use shareWithOptions because I need to pass the "to" field when sending an email.

This is my code snippet:
```typescript
SocialSharing.canShareViaEmail().then((resp)=> {
            if (resp === 'OK') {
                SocialSharing.shareViaEmail(this.body, this.subject, this.to).then((resp)=> {
                    console.log(resp);
                }).catch((err)=> {
                    console.log(err);
                });
            }
        }).catch(err=> {
            console.log(err);
        });
```
I don't see any message in the console but I see the dialog for creating the email.

Is there something I'm missing? Thanks for the help

**Comment by mhateo** [comment]
I tested in android 4.4.4 with both the gmail app and the stock email app.

**Comment by ramonornela** [comment]
I think is fix #520

**Comment by ihadeed** [comment]
Fixed in https://github.com/driftyco/ionic-native/commit/c76de34b9718144a76ee48919bb94bda956a1902 and will be available in the next release `1.3.21`

**Comment by mhateo** [comment]
Resolved by passing empty arrays in optional params.
